### ISSUE TITLE
Voltmeter mode doesn't work

### ISSUE THREAD

**Issue by PSLLSP** [clvLabs/BusPirate#1]
Voltmetr mode of ADC produced never ending series of values. It doesn't work. Well, BusPirate is switched to voltmeter mode but this Python script doesn't read values. Voltage values  are not separated by new line but rewritten in place with "0x08" control codes.

Example, test.txt:
```
d
d
D
```

Serial console, BusPirate:
```
HiZ>#
RE
Bus Pirate v3.b clone w/different PIC
Firmware v6.1 r1676  Bootloader v4.4
DEVID:0x044F REVID:0x3003 (24FJ64GA004 A3)
http://dangerousprototypes.com
HiZ>d
VOLTAGE PROBE: 0.00V
HiZ>d
VOLTAGE PROBE: 0.00V
HiZ>D
VOLTMETER MODE
Any key to exit
VOLTAGE PROBE: 0.00V
HiZ>
```
Script run with Python
```
***[ Bus Pirate scripting tool ]********************************************************************
Script file                             test.txt
COM port                                /dev/ttyUSB2
- Opening serial port
[OK] Serial port open
- Resetting board
>>> ------------------------------------| # 
<<< #
<<< RE
<<< Bus Pirate v3.b clone w/different PIC
<<< Firmware v6.1 r1676  Bootloader v4.4
<<< DEVID:0x044F REVID:0x3003 (24FJ64GA004 A3)
<<< http://dangerousprototypes.com
<<< HiZ>
- Sending script file (test.txt) - 4 lines
- Resetting board
>>> ------------------------------------| # 
<<< #
<<< RE
<<< Bus Pirate v3.b clone w/different PIC
<<< Firmware v6.1 r1676  Bootloader v4.4
<<< DEVID:0x044F REVID:0x3003 (24FJ64GA004 A3)
<<< http://dangerousprototypes.com
<<< HiZ>
>>> ------------------------------------| d 
<<< d
<<< VOLTAGE PROBE: 0.00V
<<< HiZ>
>>> ------------------------------------| d 
<<< d
<<< VOLTAGE PROBE: 0.00V
<<< HiZ>
>>> ------------------------------------| D 
<<< D
<<< VOLTMETER MODE
<<< Any key to exit

```

BTW, I miss possibility to have comments in BusPirate scripts...

**Comment by PSLLSP** [comment]
I would like to measure voltage for longer time, save results to a file and then analyse it. That task is not easy with this python because it sends many "garbage" to the output, ESC sequences to color text. It would be nice to have a switch to disable that...

A script "test1.txt" to power test circuit from AUX output and then measure voltage as fast as possible. I would like to use continuous ADC mode but because it doesn't  work, this is a workaround with single run ADC; it measures only about 300 samples, I would like to have about 50000 ;-) The limit is BusPirate buffer, I cannot send more than this, rest is just lost :-(

```
a
dddAddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
a
```

This is a simple check how stable is AUX output; note I have to force `grep` to text mode with `-a` switch:
```
$ python3 buspirate.py -c /dev/ttyUSB2 test1.txt | grep -a "VOLTAGE PROBE:" | sort | uniq -c
      3 <<< VOLTAGE PROBE: 0.00V
      1 <<< VOLTAGE PROBE: 3.25V
    250 <<< VOLTAGE PROBE: 3.26V
```

This is a signal I am interest in (color blinking LED, it changes color in a pattern and is an easy source of variable voltage values):
```
$ python3 buspirate.py -c /dev/ttyUSB2 test1.txt | grep -a "VOLTAGE PROBE:" | sort | uniq -c
      3 <<< VOLTAGE PROBE: 0.00V
     67 <<< VOLTAGE PROBE: 2.46V
    110 <<< VOLTAGE PROBE: 2.70V
     72 <<< VOLTAGE PROBE: 2.88V
      2 <<< VOLTAGE PROBE: 3.26V
```

This shows how much garbage is in the output:
```
$ python3 buspirate.py -c /dev/ttyUSB2 test1.txt | grep -a VOLTAGE | head -3 | hexdump -C
00000000  1b 5b 31 3b 33 37 6d 3c  3c 3c 20 56 4f 4c 54 41  |.[1;37m<<< VOLTA|
00000010  47 45 20 50 52 4f 42 45  3a 20 30 2e 30 30 56 0d  |GE PROBE: 0.00V.|
00000020  1b 5b 30 6d 0a 1b 5b 31  3b 33 37 6d 3c 3c 3c 20  |.[0m..[1;37m<<< |
00000030  56 4f 4c 54 41 47 45 20  50 52 4f 42 45 3a 20 30  |VOLTAGE PROBE: 0|
00000040  2e 30 30 56 0d 1b 5b 30  6d 0a 1b 5b 31 3b 33 37  |.00V..[0m..[1;37|
00000050  6d 3c 3c 3c 20 56 4f 4c  54 41 47 45 20 50 52 4f  |m<<< VOLTAGE PRO|
00000060  42 45 3a 20 30 2e 30 30  56 0d 1b 5b 30 6d 0a     |BE: 0.00V..[0m.|
0000006f
```

**Comment by clvLabs** [comment]
Hi PSLLSP, thanks for your comments!

At this moment I have my BusPirate at a friend's house... when I get it back I'll be able to check the issues you sent.
Please let me know if it's kind of urgent for you.

**Comment by clvLabs** [comment]
> BTW, I miss possibility to have comments in BusPirate scripts...

Solved with #4

**Comment by clvLabs** [comment]
> ... That task is not easy with this python because it sends many "garbage" to the output, ESC sequences to color text. It would be nice to have a switch to disable that...

Solved with #5

**Comment by clvLabs** [comment]
The `D Measure ADC (CONT.)` is special because it does not perform an action and return, as most other commands do. It continually streams data thru the serial connection until it receives a character from the user (pressed key).

Preparing this script to handle this command would force adding special syntax to the scripts to tell how much time should the program receive data until it "presses the key" to continue with the script. 

Also, all reads from the serial port are done using `Serial.readline()` and the individual voltage reads are separated by `\r` instead of `\n`, forcing to change the read method.

As a better alternative exists for continuous voltage logging using `BASIC` scripting, I'll close this as a `wontfix`.

Using `D` offers no control over timing, while BASIC could allow you to do such things as:
* Getting *bursts* of 10 reads separated by 1s once a day.
* Continuously read voltage every 100ms and ONLY display it if its greater than 3.2V
* Whatever `BASIC` allows you to do...

Please see #3 for an example of a `BASIC` voltage logging script.
